PA_Tests.cls

```diff
@@ -0,0 +1,194 @@
+		ext.getvSpecialists();
+        //ext.MyAgreement.Agreement_Type__c = 'Mugs & Bowls';
+		ext.MyAgreement.Agreement_Type__c = 'TestData';
+        ext.RefreshProds(); //calls FieldVisibilityControl()
+       // ext.queryHealthcareSpecialist(); //Called by Submit()
+
+        ext.vSpecialist = testUser.UserName;   //set the HealthSpecialist      
+        
+        //ext.getvAgreementTypes();
+        
+        ext.SumbitEmailRep(); //Calls Submit()
+		//PrdColors test input
+        test.starttest();
+        try{
+          Partner_Agreement__c vPA = [select id from Partner_Agreement__c limit 1];
+          list<Partner_Agreement_Product__c> lPAProds = [select id from Partner_Agreement_Product__c where Partner_Agreement__c= :vPA.id Order by Code__c];
+          for(integer i=0; i<lPAProds.size(); i++){
+            system.assertequals(lPAProds[i].Code__c,'TestProduct'+i+1);
+          }        	
+        }catch(exception e){
+          system.debug('Error: ' + e);
+        }
+
+        test.stoptest();        
+
+	}
+	@isTest static void test_CMDT_Other(){
+
+		Partner_Agreement__c PA = new Partner_Agreement__c();
+        PageReference pageRef = Page.PartnerAgreements;
+        Test.setCurrentPage(pageRef);
+        ApexPages.StandardController sc = new ApexPages.standardController(PA);
+        PartnerAgreement_ext ext =  new PartnerAgreement_ext(sc);
+		
+		User testUser = [select id,Name, Email, UserName from User where isActive =true and lastname = 'testing' and firstname='tester' limit 1];
+
+		ext.getvSpecialists();
+        //ext.MyAgreement.Agreement_Type__c = 'Mugs & Bowls';
+		ext.MyAgreement.Agreement_Type__c = 'TestData';
+        ext.RefreshProds(); //calls FieldVisibilityControl()
+
+		//PrdColors test input
+		list<SelectOption> options = ext.getvSpecialists();
+        test.starttest();
+        	//ext.getRepGroups();
+          map<string,Partner_Agreement_Healthcare_Specialist__c> mSpecialists = Partner_Agreement_Healthcare_Specialist__c.getAll();		    
+
+          system.assertnotequals(null,options,'Options should be instantiated');
+          system.assert(!options.isEmpty(),'Options should be populated');
+          string vlabel;
+          for(integer i=0; i<options.size(); i++){
+          vlabel = options.get(i).getvalue();
+          system.assertnotequals(null, vlabel,'Label shouldnt be empty');
+          if(vlabel == 'gwilson@WizardCast.com.test'
+            || vlabel =='pwilson@WizardCast.com.test'
+            || vlabel =='rwilson@WizardCast.com.test'|| vlabel == testUser.email 
+            ||  vlabel =='--none--' || vlabel == ''){
+
+            }else{
+              system.assert(false,vlabel +' HealthSpecialst does not match custom setting values');
+            }
+          }
+        ext.vSpecialist = testUser.Email;
+        string vUserID = ext.queryHealthcareSpecialist();
+        system.assertequals(vUserID,testUser.id);
+        test.stoptest();   
+
+	}
+	@isTest static void test_CMDT_Coverage(){
```

Comment:
We want our unit test method names to be very descriptive of what they do! we'll need to refactor this before merge so that we can quickly identify what we expected and what failed.